loader/include/Geode/utils/Task.hpp

```diff
@@ -963,10 +1030,23 @@ namespace geode {
                 MyTask::cancel(m_handle.lock());
             }
 
-            std::suspend_never initial_suspend() noexcept { return {}; }
-            std::suspend_never final_suspend() noexcept { return {}; }
-            // TODO: do something here?
-            void unhandled_exception() {}
+            std::suspend_never initial_suspend() noexcept {
+                // Queue the coroutine to start on main thread
+                queueInMainThread([this]() {
+                    std::coroutine_handle<TaskPromise>::from_promise(*this).resume();
+                });
+                return {};

```

Comment:
does this even work? youre returning suspend_never which means the coroutine will not be blocked

Reply:
my mistake

Reply:
@matcool would a suspend_always work?